gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,238 @@
+  private final String name;
+  private final String id;
+  private final Map<String, String> labels;
+  private final Long number;
+  private final State state;
+  private final Long createTimeMillis;
+  private final ResourceId parent;
+
+  public enum State {
+    LIFECYCLE_STATE_UNSPECIFIED,
+    ACTIVE,
+    DELETE_REQUESTED,
+    DELETE_IN_PROGRESS;
+  }
+
+  public static class Builder {
+    private String name;
+    private String id;
+    private Map<String, String> labels;
+    private Long number;
+    private State state;
+    private Long createTimeMillis;
+    private ResourceId parent;
+
+    Builder() {
+      labels = new HashMap<String, String>();
+    }
+
+    public Builder name(String name) {
+      this.name = name;
+      return this;
+    }
+
+    public Builder id(String id) {
+      this.id = id;
+      return this;
+    }
+
+    public Builder addLabel(String key, String value) {
+      this.labels.put(key, value);
+      return this;
+    }
+
+    public Builder removeLabel(String key) {
+      this.labels.remove(key);
+      return this;
+    }
+
+    public Builder clearLabels() {
+      this.labels.clear();
+      return this;
+    }
+
+    public Builder labels(Map<String, String> labels) {
+      this.labels = checkNotNull(labels);
+      return this;
+    }
+
+    Builder number(Long number) {
+      this.number = number;
+      return this;
+    }
+
+    Builder state(State state) {
+      this.state = state;
+      return this;
+    }
+
+    Builder createTimeMillis(Long createTimeMillis) {
+      this.createTimeMillis = createTimeMillis;
+      return this;
+    }
+
+    public Builder parent(ResourceId parent) {
+      this.parent = parent;
+      return this;
+    }
+
+    public ProjectInfo build() {
+      return new ProjectInfo(name, id, labels, number, state, createTimeMillis, parent);
+    }
+  }
+
+  ProjectInfo(String name, String id, Map<String, String> labels, Long number, State state,
+      Long createTimeMillis, ResourceId parent) {
+    this.name = name;
+    this.id = checkNotNull(id);
+    ImmutableMap.Builder<String, String> labelsMapBuilder = ImmutableMap.builder();
+    for (Map.Entry<String, String> entry : labels.entrySet()) {
+      labelsMapBuilder.put(entry.getKey(), entry.getValue());
+    }
+    this.labels = (labels == null) ? null : labelsMapBuilder.build();
+    this.number = number;
+    this.state = state;
+    this.createTimeMillis = createTimeMillis;
+    this.parent = parent;
+  }
+
+  public String id() {
+    return id;
+  }
+
+  public String name() {
+    return name;
+  }
+
+  public Long number() {
+    return number;
+  }
+
+  public Map<String, String> labels() {
+    return labels;
+  }
+
+  public State state() {
+    return state;
+  }
+
+  public Long createTimeMillis() {
+    return createTimeMillis;
+  }
+
+  public ResourceId parent() {
+    return parent;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
```

Comment:
Question: as I commented above, what do you think of using `toPb()` to implement equals?